waku/node/v2/wakunode2.nim

```diff
@@ -300,9 +295,9 @@ when isMainModule:
       Port(uint16(conf.tcpPort) + conf.portsShift),
       Port(uint16(conf.udpPort) + conf.portsShift))
     node = WakuNode.init(conf.nodeKey, conf.libp2pAddress,
-      Port(uint16(conf.tcpPort) + conf.portsShift), extIp, extTcpPort, conf.topics.split(" "))
+      Port(uint16(conf.tcpPort) + conf.portsShift), extIp, extTcpPort)
 
-  waitFor node.start()
+  waitFor node.start(conf.topics.split(" "))
```

Comment:
E.g. instead do it here and do it after static node etc has been connected to.

Reply:
Speaking of, I suppose this how the code actually executes. First the connect to nodes is called, then async we are waiting until we subscribe. It'd be easier to read and be more separated in terms of concerns if it was there

Reply:
The downside of doing it in main I suppose is that some applications will use it as a library and then it wouldn't be called automatically. Honestly I wonder if that isn't fine. The reason we are doing this in first place is to get nodes in cluster to subscribe to some topics by default so machine can do proper relaying.

@kdeme wdyt?

Reply:
Well, if we do want to have this done here, some helper call should be created. Realize also that a node will not necessarily start `WakuRelay` (but instead e.g. only WakuFilter), so some extra logic will eventually still be needed.

Reply:
Agree re helper function. And yes, the FilterRequest would be an enhancement to do as well.

The reason for this default change in the first place is to allow for "non human operated" nodes like cluster node. Without it it doesn't work. I think there's diminishing returns for providing defaults when there's an API that should be used in standard operation.

Reply:
does the absence of a response here means you fixed it?

Reply:
@oskarth this isn't really relevant anymore. now that we mountRelay the topic subscriptions are there.